src/components/AddressInput.vue

```diff
@@ -0,0 +1,394 @@
+    }
+
+    private _updateSelection() {
+        const textarea = this.$refs.textarea;
+        const focused = document.activeElement === textarea
+            && (textarea.selectionStart !== AddressInput.ADDRESS_MAX_LENGTH // if all blocks are filled and the caret
+            || textarea.selectionEnd !== AddressInput.ADDRESS_MAX_LENGTH); // is at the end display as if not focused
+        const selectionStartBlock = focused ? Math.floor(textarea.selectionStart / 5) : -1;
+        const selectionEndBlock = focused ? Math.floor(textarea.selectionEnd / 5) : -1;
+        if (selectionStartBlock === this.selectionStartBlock && selectionEndBlock === this.selectionEndBlock) return;
+        this.selectionStartBlock = selectionStartBlock;
+        this.selectionEndBlock = selectionEndBlock;
+        this._updateTextClipBackground();
+    }
+
+    private _updateTextClipBackground() {
+        const textarea = this.$refs.textarea;
+        if (textarea.classList.contains('background-clip-text-unsupported')) return;
+
+        if (!textarea.style.backgroundImage) {
+            // This is the first time we want to set a background. First check for support.
+            const computedStyle = window.getComputedStyle(textarea);
+            const supported = computedStyle['caret-color']
+                && (computedStyle['background-clip'] === 'text' || computedStyle['-webkit-background-clip'] === 'text');
+            textarea.classList.toggle('background-clip-text-unsupported', !supported);
+            if (!supported) return;
+        }
+
+        if (textarea.value.length === 0) {
+            // placeholder is shown
+            textarea.style.backgroundImage = 'unset';
+            textarea.style.backgroundColor = 'unset';
+            return;
+        }
+
+        // Generate a svg background image similar to the mask image that sets the color for focused blocks.
+        // Chrome (and potentially others) only rerender the background-clip on background change, not on text change,
+        // therefore we assign an incremental id to ensure a new svg is generated each time. In Firefox we don't need
+        // that and do not generate new svgs every time to reduce flickering.
+        const svgId = !BrowserDetection.isFirefox() ? `text-clip-${Date.now()}` : `text-clip`;
+        let svg = `<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 220 123" id="${svgId}">`;
+        const xOffset = 0;
+        const yOffset = 6;
+        const xStep = 76;
+        const yStep = 41;
+        for (let row = 0; row < 3; ++row) {
+            for (let column = 0; column < 3; ++column) {
+                const blockIndex = 3 * row + column;
+                if (blockIndex < this.selectionStartBlock || blockIndex > this.selectionEndBlock) continue;
+                svg += `<rect x="${xOffset + column * xStep}" y="${yOffset + row * yStep}" `
+                       + 'fill="#0582ca" width="59" height="27"/>'; // nimiq-light-bue
```

Comment:
I think if you don't use a `#` hash character here for the fill color, but it's URL representation `%23`, you don't have to base64-encode the svg.

Reply:
I had tried that first and also expected it to work but it didn't.
I don't know what else would be needed to make it work.
But base64 works out of the box, so that's fine.